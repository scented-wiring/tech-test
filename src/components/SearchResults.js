import React from "react";
import "../styles/SearchResults.css";

const SearchResults = ({ results }) => {
  if (!results.length) {
    return (
      <div className="search-results">
        <p>No results.</p>
      </div>
    );
  } else {
    return (
      <div id="images-wrapper">
        {results.map((image) => (
          <img className="card-image" src={image} alt="spaceImage" />
        ))}
      </div>
    );
  }
};

export default SearchResults;
