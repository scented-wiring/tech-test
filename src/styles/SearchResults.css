.search-results {
  text-align: center;
}

.search-results-image {
  width: 30%;
  height: 30%;
}
