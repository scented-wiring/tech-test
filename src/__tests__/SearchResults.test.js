import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const { asFragment } = render(<SearchResults />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
