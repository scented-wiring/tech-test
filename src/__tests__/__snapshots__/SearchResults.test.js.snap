// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    id="images-wrapper"
  >
    <a
      href="[object Object]"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="spaceImage"
        class="card-image"
        src="[object Object]"
      />
    </a>
  </div>
</DocumentFragment>
`;
